### PR TITLE
Add jsdoc to the project and generate docs for the player API

### PR DESCRIPTION
**Pull request by robwalch**
### This PR will...

Define the player API as a class which includes all public API members and methods. Added annotations for members and methods we intend to support. Items we plan to deprecate in v8.0 will be moved to a compatibility script.

### Why is this Pull Request needed?

This is to make maintenance of our API and developer docs easier, to make both more consistent, and to provide another source of reference for developers using the player API.

### To consider

Does it makes sense to include the generated docs in each build and source control? It looks like having a docs/ folder in the master branch will make it easy to publish to docs on github https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/

You can pull down this branch and run `npm run docs` to review the generated pages.

#### Addresses Issue(s):

JW8-108

### CODE REVIEW

File: src/js/api/api.js [hunk 1]
```diff
@@ -1,266 +1,1210 @@
...
-        _qoe.tick('init');
+            let _controller = setupController(this, element);
 
+            /**
+             * Call an internal method on the player's controller.
+             * @param {string} name - The method to call.
+             * @param {...*} [args] - Any arguments made after the name are passed to the internal method.
+             * @return {any} Returns the result or null if the method is undefined.
+             */
+            this.callInternal = function(name, ...args) {
```
Comment (johnBartos) [inline src/js/api/api.js]: `@return {*}`

File: src/js/playlist/item.js [hunk 2]
```diff
@@ -9,6 +9,18 @@ define([
         minDvrWindow: 120
     };
 
+    /**
+     * An item in the playlist
+     * @typedef {object} PlaylistItem
+     * @property {Array.<PlaylistItemSource>} sources - A list of alternative media sources for the player to choose from.
+     * @property {Array.<PlaylistItemTrack>} tracks - A list tracks associated with this item.
```
Comment (karimJWP) [inline src/js/playlist/item.js]: list *of* tracks

File: src/js/api/api.js [hunk 3]
```diff
@@ -1,266 +1,1223 @@
...
-            }
-            args.type = type;
-            var jwplayer = window.jwplayer;
-            if (jwplayer && jwplayer.debug) {
-                return Events.trigger.call(_this, type, args);
+        return controller;
+    }
+
+    /**
+     * Detach Api events listeners and destroy the controller.
```
Comment (johnBartos) [inline src/js/api/api.js]: Detach`es` API event~s~ listeners and destroy`s` the controller

File: src/js/api/api.js [hunk 4]
```diff
@@ -1,266 +1,1223 @@
...
+        api.off();
+        controller.off();
+        // so players can be removed before loading completes
+        if (controller.playerDestroy) {
+            controller.playerDestroy();
+        }
+    }
+
+    /**
+     * Remove the Api instance from the list of active players.
```
Comment (johnBartos) [inline src/js/api/api.js]: Remove`s`

File: src/js/api/api.js [hunk 5]
```diff
@@ -1,266 +1,1223 @@
...
+    /** Class representing the jwplayer() API
+     */
+    class Api {
 
-            // capture the ready event and add setup time to it
-            _controller.on(events.JWPLAYER_READY, function(event) {
-                _qoe.tick('ready');
-                event.setupTime = _qoe.between('setup', 'ready');
+        /**
+         * Create a player instance.
```
Comment (johnBartos) [inline src/js/api/api.js]: Create`s`

**Pull request merged**